crates/task/src/lib.rs

```diff
@@ -199,14 +197,42 @@ impl TaskVariables {
             }
         })
     }
+
+    /// Returns iterator over names of all set task variables.
+    pub fn keys(&self) -> impl Iterator<Item = &VariableName> {
+        self.0.keys()
+    }
 }
 
 impl FromIterator<(VariableName, String)> for TaskVariables {
     fn from_iter<T: IntoIterator<Item = (VariableName, String)>>(iter: T) -> Self {
-        Self(HashMap::from_iter(iter))
+        Self(BTreeMap::from_iter(iter))
     }
 }
 
+impl FromStr for VariableName {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let without_prefix = s.strip_prefix(ZED_VARIABLE_NAME_PREFIX).ok_or(())?;
+        let value = match without_prefix {
+            "FILE" => Self::File,
```

Comment:
We may wan to also add the `STEM` and `FILENAME` when #12017 gets merged.

Reply:
fyi, #12245 just added these as `ZED_DIRNAME` and `ZED_RELATIVE_FILE`.